File: backend/app/schemas/reports.py
"""Reports aggregated response schemas — matches FE ReportsData interface."""

from app.schemas.base import CamelModel


class PeriodSummary(CamelModel):
    income: float
    expense: float
    savings: float
    savings_rate: float


class CategoryComparison(CamelModel):
    category_id: str
    name: str
    icon: str
    color: str
    current: float
    previous: float
    pct: float


class ChartPoint(CamelModel):
    label: str
    income: float
    expense: float


class TrendPoint(CamelModel):
    label: str
    cum_expense: float
    expense: float


class CashFlowData(CamelModel):
    income_items: list[CategoryComparison]
    expense_items: list[CategoryComparison]
    total_income: float
    total_expense: float
    net: float


class TopTransaction(CamelModel):
    type: str
    amount: float
    date: str
    category_name: str
    category_icon: str
    category_color: str


class ReportsResponse(CamelModel):
    current: PeriodSummary
    previous: PeriodSummary
    chart_data: list[ChartPoint]
    trend_data: list[TrendPoint]
    expense_by_category: list[CategoryComparison]
    income_by_category: list[CategoryComparison]
    cash_flow: CashFlowData
    top_transactions: list[TopTransaction]

File: frontend/types/reports.ts
export interface PeriodSummary {
  income: number
  expense: number
  savings: number
  savingsRate: number
}

export interface CategoryComparison {
  categoryId: string
  name: string
  icon: string
  color: string
  current: number
  previous: number
  pct: number
}

export interface ChartPoint {
  label: string
  income: number
  expense: number
}

export interface TrendPoint {
  label: string
  cumExpense: number
  expense: number
}

export interface CashFlowData {
  incomeItems: CategoryComparison[]
  expenseItems: CategoryComparison[]
  totalIncome: number
  totalExpense: number
  net: number
}

export interface TopTransaction {
  type: string
  amount: number
  date: string
  description: string
  categoryName: string
  categoryIcon: string
  categoryColor: string
}

export interface ReportsData {
  current: PeriodSummary
  previous: PeriodSummary
  chartData: ChartPoint[]
  trendData: TrendPoint[]
  expenseByCategory: CategoryComparison[]
  incomeByCategory: CategoryComparison[]
  cashFlow: CashFlowData
  topTransactions: TopTransaction[]
}
